components/elm/src/biogeophys/ActiveLayerMod.F90

```diff
@@ -144,14 +160,111 @@ subroutine alt_calc(num_soilc, filter_soilc, &
             endif
          endif
 
-
          ! if appropriate, update maximum annual active layer thickness
          if (alt(c) > altmax(c)) then
             altmax(c) = alt(c)
             altmax_indx(c) = alt_indx(c)
          endif
+         if (alt(c) > altmax_ever(c)) then
+            if (spinup_state .eq. 0) then
+                altmax_ever(c) = alt(c)
+                altmax_ever_indx(c) = alt_indx(c)
+            else !overwrite if in spinup
+                altmax_ever(c) = 0._r8
+                altmax_ever_indx(c) = 0
+            endif
+         endif
+
+         ! special loop for if year = 1989, see above note regarding
+         ! replacing with more physically based mechanism.
+         if (year .eq. 1989) then
+            altmax_1989(c) = altmax(c)
+            altmax_1989_indx(c) = altmax_indx(c)
+         endif
```

Comment:
Could this all be moved within the `if (use_polygonal_tundra) then` at line 185?

Reply:
Moved as suggested